bc_obps/reporting/service/report_supplementary_version_service.py

```diff
@@ -469,31 +473,52 @@ def clone_report_version_facility_product_data(
             ReportSupplementaryVersionService.clone_report_version_facility_product_emission_allocations(
                 old_report_version=old_report_version,
                 new_report_version=new_report_version,
-                old_facility_report=old_facility_report_to_clone,
                 new_facility_report=new_facility_report,
                 old_report_product_to_clone=old_report_product_to_clone,
                 new_report_product=cloned_report_product,
             )
 
     @staticmethod
-    def clone_report_version_facility_product_emission_allocations(
+    def clone_report_version_facility_emission_allocation(
         old_report_version: ReportVersion,
         new_report_version: ReportVersion,
         old_facility_report: FacilityReport,
         new_facility_report: FacilityReport,
+    ) -> None:
+        old_emission_allocation = ReportEmissionAllocation.objects.filter(
+            report_version_id=old_report_version,
+            facility_report=old_facility_report,
+        ).first()
+        if old_emission_allocation:
+            cloned_allocation = copy.deepcopy(old_emission_allocation)
+            cloned_allocation.pk = None
+            cloned_allocation.report_version = new_report_version
+            cloned_allocation.facility_report = new_facility_report
+            cloned_allocation.save()
```

Comment:
we don't have to use `copy.deepcopy` here, there is an official documented way to duplicate django models: https://docs.djangoproject.com/en/5.0/topics/db/queries/#copying-model-instances

instead, you can do:
```suggestion
        emission_allocation_to_copy = ReportEmissionAllocation.objects.filter(
            report_version_id=old_report_version,
            facility_report=old_facility_report,
        ).first()
        if emission_allocation_to_copy:
            emission_allocation_to_copy.pk = None
            cloned_allocation.save()
```

and it will save a new object since the pk is None